test/apiserver_organization_test.go

```diff
@@ -122,8 +123,22 @@ modfor:
 	}
 
 	t.Log("update")
-	org.Labels = map[string]string{"aa": "bb"}
-	assert.NoError(t, cl.Update(ctx, org))
+	updateCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	t.Cleanup(cancel)
+	require.NoError(t, wait.PollUntil(time.Second, func() (done bool, err error) {
```

Comment:
what bout using `controllerutil.CreateOrUpdate` with retry mechanism on version conflict? This logic should be extracted in a separate function. 

Why I insist on extracting this away from the main test function is test readability. This huge poll block distracts from it purpose, which is updating the object and handling the modified errors. Let me know if we need a call or something to discuss this further.

Reply:
If you check the readability of the e2e tests... there are some blocks are dropping the readability, for example 3 similar blocks reading stuff from `ResultChan`
I don't really see my few lines of Update can drop the readability of e2e tests

and why should I use createOrUpdate? it is get and then update, nothing about create

Reply:
Done